src/components/interactive-builder/edit-question.modal.tsx

```diff
@@ -397,57 +401,63 @@ const EditQuestionModal: React.FC<EditQuestionModalProps> = ({
               )}
               required
             />
-            <RadioButtonGroup
-              defaultSelected={/true/.test(questionToEdit?.required?.toString()) ? 'required' : 'optional'}
-              name="isQuestionRequired"
-              legendText={t(
-                'isQuestionRequiredOrOptional',
-                'Is this question a required or optional field? Required fields must be answered before the form can be submitted.',
-              )}
-            >
-              <RadioButton
-                id="questionIsNotRequired"
-                defaultChecked={true}
-                labelText={t('optional', 'Optional')}
-                onClick={() => setIsQuestionRequired(false)}
-                value="optional"
-              />
-              <RadioButton
-                id="questionIsRequired"
-                defaultChecked={false}
-                labelText={t('required', 'Required')}
-                onClick={() => setIsQuestionRequired(true)}
-                value="required"
-              />
-            </RadioButtonGroup>
-            <Select
-              defaultValue={questionToEdit.type}
-              onChange={(event: React.ChangeEvent<HTMLSelectElement>) =>
-                setQuestionType(event.target.value as QuestionType)
-              }
-              id={'questionType'}
-              invalidText={t('typeRequired', 'Type is required')}
-              labelText={t('questionType', 'Question type')}
-              required
-            >
-              {!questionType && <SelectItem text={t('chooseQuestionType', 'Choose a question type')} value="" />}
-              {questionTypes.map((questionType, key) => (
-                <SelectItem text={questionType} value={questionType} key={key} />
-              ))}
-            </Select>
-            <Select
-              defaultValue={questionToEdit.questionOptions.rendering}
-              onChange={(event: React.ChangeEvent<HTMLSelectElement>) => setFieldType(event.target.value as RenderType)}
-              id="renderingType"
-              invalidText={t('validFieldTypeRequired', 'A valid field type value is required')}
-              labelText={t('renderingType', 'Rendering type')}
-              required
-            >
-              {!fieldType && <SelectItem text={t('chooseRenderingType', 'Choose a rendering type')} value="" />}
-              {fieldTypes.map((fieldType, key) => (
-                <SelectItem text={fieldType} value={fieldType} key={key} />
-              ))}
-            </Select>
+            {'label' in questionToEdit && (
```

Comment:
Since every question has a label, although these fields are not required for markdown questions, they would get rendered, which doesn't happen in add-question-modal. We should change this condition to match the one we used in the add-question-modal
<img width="1710" alt="Screenshot 2024-11-18 at 11 39 14 AM" src="https://github.com/user-attachments/assets/bc2015cc-f686-4e11-aca7-800708e08041">
<img width="1710" alt="Screenshot 2024-11-18 at 11 41 04 AM" src="https://github.com/user-attachments/assets/82790ac3-18c9-47d2-9d6c-90da0344521c">